tests/component/test_stream_writers_do.py

```diff
@@ -0,0 +1,615 @@
+import ctypes
+import math
+from typing import Callable, List
+
+import numpy
+import pytest
+
+import nidaqmx
+from nidaqmx.constants import LineGrouping
+from nidaqmx.stream_writers import DigitalMultiChannelWriter, DigitalSingleChannelWriter
+from nidaqmx.utils import flatten_channel_string
+
+
+def _start_do_task(task: nidaqmx.Task, is_port: bool = False, num_chans: int = 1) -> None:
+    # We'll be doing on-demand, so start the task and drive all lines low
+    task.start()
+    if is_port:
+        if num_chans == 8:
+            task.write(0)
+        else:
+            task.write([0 for _ in range(num_chans)])
+    else:
+        if num_chans == 1:
+            task.write(False)
+        else:
+            task.write([False for _ in range(num_chans)])
+
+
+@pytest.fixture
+def do_single_line_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    task.do_channels.add_do_chan(
+        real_x_series_device.do_lines[0].name, line_grouping=LineGrouping.CHAN_FOR_ALL_LINES
+    )
+    _start_do_task(task)
+    return task
+
+
+@pytest.fixture
+def do_single_channel_multi_line_task(
+    generate_task: Callable[[], nidaqmx.Task], real_x_series_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task = generate_task()
+    chan = task.do_channels.add_do_chan(
+        flatten_channel_string(real_x_series_device.do_lines[:8].name),
```

Comment:
`PhysicalChannel.name` should be a str, not a list. You are hitting this bug: https://github.com/ni/nidaqmx-python/issues/392

As a workaround, I suggest `flatten_channel_string(real_x_series_device.do_lines.channel_names[:8])`

Or you could fix the bug 😁

Reply:
I'll workaround for now, and then fix that once I'm done with my set of new component tests